lsp-test/src/Language/LSP/Test.hs

```diff
@@ -421,7 +421,7 @@ sendRequest method params = do
   reqMap <- requestMap <$> ask
   liftIO $
     modifyMVar_ reqMap $
-      \r -> return $ fromJust $ updateRequestMap r id method
+      \r -> return $ fromMaybe r $ updateRequestMap r id method
```

Comment:
updateRequestMap returns Nothing when request with given LspId is already it.
I don't see how it makes sense to crash in such cases as opposed to returning the origial map unmodified. Similarly in the file below.

Reply:
It's quite bad: that means we've somehow reused an ID which means stuff will break. So I think this probably _should_ throw, but maybe not by using `fromJust`

Reply:
Alright I'll play around with it locally to see if I can reproduce the  me play around with how this works to see if I can reproduce the `  Timed out waiting to receive a message from the server.` locally and then get back to this or close this.